### PR TITLE
test: fix test_ec2_ipv6

### PR DESCRIPTION
**Pull request by aciba90**
## Proposed Commit Message

```
test: fix test_ec2_ipv6
    
The test removes dhcp clients to force imds recrawling, but didn't
remove the dhcpcd one, after 21b2b6e4423b0fec325b32042c05ef4274cdd301
the dhcpcd one is used and the test preconditoin is not met.
    
Co-authored-by: Chad Smith <chad.smith@canonical.com>
```

## Additional Context

https://jenkins.canonical.com/server-team/view/cloud-init/job/cloud-init-integration-mantic-ec2/139

## Test Steps

```sh
export CLOUD_INIT_CLOUD_INIT_SOURCE=ppa:cloud-init-dev/daily
export CLOUD_INIT_PLATFORM=ec2
export CLOUD_INIT_OS_IMAGE=mantic

tox -e integration-tests -- --pdb tests/integration_tests/datasources/test_ec2_ipv6.py::test_dual_stack
```

## Checklist

- [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/development/index.html)
- [ ] I have updated or added any [unit tests](https://cloudinit.readthedocs.io/en/latest/development/testing.html) accordingly
- [ ] I have updated or added any [documentation](https://cloudinit.readthedocs.io/en/latest/development/contribute_docs.html) accordingly

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

File: tests/integration_tests/datasources/test_ec2_ipv6.py [hunk 1]
```diff
@@ -41,10 +41,14 @@ def test_dual_stack(client: IntegrationInstance):
...
 
-    # Force NoDHCPLeaseError (by removing dhclient) and assert ipv6 still works
+    # Force NoDHCPLeaseError (by removing dhcp clients) and assert ipv6 still
+    # works
     # Destructive test goes last
     # dhclient is at /sbin/dhclient on bionic but /usr/sbin/dhclient elseware
-    assert client.execute("rm $(which dhclient)").ok
+    assert client.execute("rm $(command -v dhclient)").ok
+    # Not preinstalled on all versions
+    client.execute("rm $(command -v dhcpcd)")
```
Comment (blackboxsw) [inline tests/integration_tests/datasources/test_ec2_ipv6.py:50]: I think we want to only attempt to perform this on Mantic ++ right?
So should we assert we removed this explicitly with the following?

```suggestion
    if CURRENT_RELEASE > MANTIC:
        client.execute("rm $(command -v dhcpcd)").ok
    fi
```

We'd need an import
`from tests.integration_tests.releases import CURRENT_RELEASE, MANTIC`

**Review by blackboxsw — CHANGES_REQUESTED** [review]
Minor change requested to only invoke the `rm <client>` on systems which have dhclient and/or dhcpcd.

File: tests/integration_tests/datasources/test_ec2_ipv6.py (hunk 1 above)
Comment (blackboxsw) [reply tests/integration_tests/datasources/test_ec2_ipv6.py:50]: I realize that this review suggestion is basically the same thing as your proposal.  But, it does allow us to assert that we properly removed the dhcp_client as opposed to ignoring the exit code of the rm operation in your approach.

Comment (aciba90) [reply tests/integration_tests/datasources/test_ec2_ipv6.py]: done

**Comment by aciba90** [comment]
Thanks, @blackboxsw, for the review. Suggestion applied.

**Review by blackboxsw — APPROVED** [review]
+1 thanks!

**Pull request merged**